fontir/src/orchestration.rs

```diff
@@ -0,0 +1,108 @@
+// Helps coordinate the graph execution
+// Eventually has to move outside fontir to let us use for BE work
+
+use std::{
+    collections::{HashMap, HashSet},
+    sync::Arc,
+};
+
+use parking_lot::RwLock;
+
+use crate::ir::{GlobalMetadata, GlyphIr};
+
+// Unique identifier of work. If there are no fields work is unique.
+#[derive(Copy, Clone, Debug, PartialEq, Eq, Hash)]
+pub enum WorkIdentifier {
+    GlobalMetadata,
+    GlyphIr(u32),
+    FinishIr,
+}
+
+const MISSING_DATA: &str = "Missing data, dependency management failed us?";
+
+/// Read/write access to data for async work.
+///
+/// Intent is a root orchestrator creates a context and share copies with restricted
+/// access with spawned tasks. Copies with access control are created to detect bad
+/// execution order / mistakes, not to block actual bad actors.
+pub struct Context {
+    // If present, the one and only key you are allowed to write to
+    // Otherwise you totally get to write whatever you like
+    write_mask: Option<WorkIdentifier>,
+
+    // If present, what you can access through this context
+    // Intent is root has None, task-specific Context only allows access to dependencies
+    read_mask: Option<HashSet<WorkIdentifier>>,
+
+    // work results we've completed or restored from disk
+    // We create individual caches so we can return typed results from get fns
+    global_metadata_cache: Arc<RwLock<Option<Arc<GlobalMetadata>>>>,
```

Comment:
This feels like it might want to be its own type.

Reply:
Heh +1!

Reply:
Hm. OK, only a very shallow type for now as the types cached vary. If I find strong patterns as we add more we can add more types.